sql/expression/tablefunction/table_function.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dtablefunctions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+var _ sql.TableFunction = &TableFunction{}
+var _ sql.ExecSourceRel = &TableFunction{}
+
+type TableFunction struct {
```

Comment:
How about a name that describes more that this is wrapper around a regular function to turn it into a table function? That would help someone looking at this code in the future figure out the intent of this type faster. Maybe something simple like `TableFunctionWrapper` or `TableFunctionAdapter`?